pulsar/message_chunking_test.go

```diff
@@ -576,3 +548,27 @@ func createTestMessagePayload(size int) []byte {
 	}
 	return payload
 }
+
+//nolint:all
+func sendSingleChunk(p Producer, uuid string, chunkID int, totalChunks int) {
+	msg := &ProducerMessage{
+		Payload: []byte(fmt.Sprintf("chunk-%s-%d|", uuid, chunkID)),
+	}
+	producerImpl := p.(*producer).producers[0].(*partitionProducer)
+	mm := producerImpl.genMetadata(msg, len(msg.Payload), time.Now())
+	mm.Uuid = proto.String(uuid)
+	mm.NumChunksFromMsg = proto.Int32(int32(totalChunks))
+	mm.TotalChunkMsgSize = proto.Int32(int32(len(msg.Payload)))
+	mm.ChunkId = proto.Int32(int32(chunkID))
+	producerImpl.updateMetadataSeqID(mm, msg)
+
+	producerImpl.internalSingleSend(
+		mm,
+		msg.Payload,
+		&sendRequest{
+			callback: func(id MessageID, producerMessage *ProducerMessage, err error) {},
+			msg:      msg,
+		},
+		uint32(internal.MaxMessageSize),
+	)
```

Comment:
We need to flush the producer after sending the message. Or handle the callback of the `sendRequest`.

Reply:
> We need to flush the producer after sending the message

OK, I will fix it.